Jumpscale/sal/hostfile/tests/base.py

```diff
@@ -0,0 +1,17 @@
+from Jumpscale import j
+
+
+def main(self):
+    """
+    to run:
+
+    js_shell 'j.sal.hostfile._test(name="base")'
+
+    """
+
+    j.sal.process.execute('echo "194.45.24.74 test" >> /etc/hosts')
+    assert j.sal.hostsfile.ip_exists('194.45.24.74') is True
+    j.sal.hostsfile.hostnames_set('194.45.24.74', ['testhostname'])
+    assert 'testhostname' in j.sal.hostsfile.hostnames_get('194.45.24.74')
+    j.sal.hostsfile.ip_remove('194.45.24.74')
+    assert j.sal.hostsfile.ip_exists('194.45.24.74') is False
```

Comment:
I would do some cleanup in a finally clause, and also backup the original file and restore it. The test at the moment have a lot of side-effects.

Reply:
if only some test framework with proper setup/teardown would exists.. (https://docs.pytest.org/en/latest/) ;-)